drivers/mtd_flashpage/mtd_flashpage.c

```diff
@@ -65,18 +82,18 @@ static int _write(mtd_dev_t *dev, const void *buf, uint32_t addr, uint32_t size)
         return -EINVAL;
     }
 #endif
+    assert(CPU_FLASH_BASE % FLASHPAGE_WRITE_BLOCK_ALIGNMENT);
```

Comment:
@jue89 : Not sure why this needed to be added. Is this something that should be asserted elsewhere (maybe in the `flashpage` driver)?

Reply:
This assert just makes sure that just testing the `addr` (which is an offset starting from CPU_FLASH_BASE) for alignment is a valid thing to do. Otherwise we must check `CPU_FLASH_BASE + addr` for alignment during runtime.

I asserted this at this point because it checks assumptions made one line later.

Reply:
Ok, I suggest making it a `static_assert` then, and maybe a comment to explain its purpose. If you agree, I will make the change.